src/Services/Implementations/ReverseGeocodes/OpenStreetMapReverseGeocodeServiceBase.cs

```diff
@@ -33,12 +40,21 @@ public async Task<IEnumerable<string>> Get(Coordinate coordinate, List<PropertyI
 		return addressPropertyValueList;
 	}
 
-	public async Task<OpenStreetMapResponse?> SerializeFullResponse(Coordinate coordinate)
+	public async Task<OpenStreetMapResponse?> SerializeFullResponse(ReverseGeocodeRequest request)
 	{
 		try
 		{
-			var requestUri = RequestUri(coordinate);
+			var requestUri = RequestUri(request.Coordinate.Latitude, request.Coordinate.Longitude);
+
+			if (_cache.TryGetValue(request, out OpenStreetMapResponse? cachedData))
+			{
+				return cachedData;
```

Comment:
This code is very straightforward easily testable by manual tests if the same coordinate bucket is requested from cache. It will be perfect to write an unit test to verify this cache hit logic.

This unit test will also cover your ReverseGeocodeRequest equality and it's hash code implementation.

Creating an new unit test on OpenStreetMapReverseGeocodeServiceUnitTests just to cover cache hit & miss scenarios would be perfect.

Reply:
The constructor method of OpenStreetMapReverseGeocodeServiceBase's access modifier is currently protected, preventing the creation of a specific test for its SerializeFullResponse method. Is it appropriate to modify the access modifier of this constructor to public for testing purposes? @alpcoker

Reply:
Yes, sure.